test/rules/blas.jl

```diff
@@ -20,6 +20,41 @@ using LinearAlgebra.BLAS: gemm
                 x̄_ad = dx(ȳ)
                 x̄_fd = j′vp(central_fdm(5, 1), f, ȳ, x)
                 @test x̄_ad ≈ x̄_fd rtol=1e-9 atol=1e-9
+                if size(x) != ()  # A and B
+                    @test dx isa Rule{<:Function,<:Function}
+                    x̄ = zeros(size(x)...)
```

Comment:
Could we please implement helper functionality similar to that found in Nabla.jl's tests for these accumulation rules? We're going to be adding quite a few of these `accumulate!` rules, so would be worthwhile doing now, with this PR, rather than later on.

Reply:
We should be able to use `rrule_test` here, which includes accumulation testing, once #56 is merged. Currently `rrule_test` can't be used because it assumes all arguments are differentiable, which isn't the case for the BLAS functions.